src/NuGet.Core/NuGet.Frameworks/AssetTargetFallbackFramework.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Shared;
+
+#if IS_NET40_CLIENT
+using FallbackList = System.Collections.Generic.IList<NuGet.Frameworks.NuGetFramework>;
+#else
+using FallbackList = System.Collections.Generic.IReadOnlyList<NuGet.Frameworks.NuGetFramework>;
+#endif
+
+namespace NuGet.Frameworks
+{
+    /// <summary>
+    /// AssetTargetFallbackFramework only fallback when zero assets are selected. These do not 
+    /// auto fallback during GetNearest as FallbackFramework would.
+    /// </summary>
+#if NUGET_FRAMEWORKS_INTERNAL
```

Comment:
What's up? This is so NuGet.Frameworks can be compiled into NuGet.Core without exposing new API surface area.